### ISSUE TITLE
Support iOS 18 Dark and Tinted Icons

### ISSUE THREAD

**Issue by bbb651** [hrydgard/ppsspp#19475]
### What should happen

ppsspp should support iOS 18’s new icon format that allows for dark and tinted icons. See this [HIG page](https://developer.apple.com/design/human-interface-guidelines/app-icons#iOS-iPadOS) for more information. The icon already is already made up of an dpad and a solid background, so separating them to the different layers should be a good starting point. To better match the design guideline maybe the dpad border could be removed in the dark variant and instead have the dpad gradient span the entire dpad rather than individual buttons?

It’s worth noting that tinted icons remove the difference between the ppsspp and ppsspp gold icons, although tinted icons are forced even for unsupported apps when used so afaik it’s not a concern for this.

### Who would this benefit

All iOS 18 users who make use of these features

### Platform (if relevant)

iOS

### Games this would be useful in

N/A

### Checklist

- [X] Check the [latest git build](https://www.ppsspp.org/devbuilds) in case it's already implemented.
- [X] [Search for other requests](https://github.com/hrydgard/ppsspp/search?q=my+issue&type=issues) of the same feature.

**Comment by anr2me** [comment]
Hmm.. since tinted icon is provided as a grayscale, it will looked the same on both gold and normal version, unless each one have a slightly different shape.

**Comment by bbb651** [comment]
How it could look like:
![image](https://github.com/user-attachments/assets/72eb90a3-0a4a-4866-962c-7b990345d547)
I tried to make a borderless one-gradient version of the icon but it didn't turn out that great: 
![image](https://github.com/user-attachments/assets/ad0be260-bba6-4f83-a4f4-85abf21dcf90)
(I don't have a macOS development setup to test it but I'll try to make a PR)

**Comment by hrydgard** [comment]
There is a small shape difference between regular and gold, the "glints" (highlights).

Anyway, I don't really understand how this is different from the existing icons. Can't the OS already tint them?

**Comment by bbb651** [comment]
I made a PR #19476, I noticed the different gradients and glints. I referenced [Unwatched](https://github.com/fer0n/Unwatched/tree/main/Unwatched/Assets.xcassets/icons/AppIcon.appiconset) when making it.

See the [HIG page](https://developer.apple.com/design/human-interface-guidelines/app-icons#iOS-iPadOS), specifically this section:
![image](https://github.com/user-attachments/assets/38fa4aac-97f7-4271-afeb-e811080a8470)

It automatically converts some apps icons to dark mode (it needs to remove the background for it to work), and all apps to tinted, not sure what heuristic it uses (my guess is that it checks if a certain margin around the icon is the exact same color to determine the background and the dpad is slightly too large to it to work, or the shadow/anti-aliasing trips it up).
